wit/types.wit

```diff
@@ -1,44 +1,46 @@
-interface messaging-types {
+interface types {
     /// A connection to a message-exchange service (e.g., buffer, broker, etc.).
     resource client {
         connect: static func(name: string) -> result<client, error>;
     }
     
-    /// TODO(danbugs): This should be eventually extracted as an underlying type for other wasi-cloud-core interfaces.
-    resource error {
-        trace: static func() -> string;    
+    /// Errors that can occur when using the messaging interface.
+    variant error {
+        /// The requested option is not authorized. This could be a topic it doesn't have
+        /// permission to subscribe to, or a permission it doesn't have to perform a specific
+        /// action. This error is mainly used when calling `set-subscriptions` on a guest.
+        unauthorized,
+        /// The request or operation timed out.
+        timeout,
+        /// An error occurred with the connection. Includes a message for additional context
+        connection(string),
+        /// Work on the message was abandoned for the given reason
+        abandoned(string),
+        /// A catch all for other types of errors
+        other(string),
     }
   
-    /// There are two types of channels:
-    /// - publish-subscribe channel, which is a broadcast channel, and
-    /// - point-to-point channel, which is a unicast channel.
-    ///
-    /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
-    type channel = string;
-  
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
-    record guest-configuration {
-        channels: list<channel>,
-        extensions: option<list<tuple<string, string>>>
-    }
-  
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
+        constructor(topic: string, data: list<u8>);
```

Comment:
Please note that I shrunk this down to avoid any breaking changes. I can't think of any scenario where you wouldn't need to provide at least a topic and a body (though sometimes that body would be empty, so I could get behind an argument we don't need that parameter, but I still think it is useful to have). You can still set the other data members using the getters and the setters